### PR TITLE
Plugin chosen improvements

### PR DESCRIPTION
**Pull request by mokaddem**
**Merge request checklist**

- [X] I read the [guidelines for contributing](https://github.com/mistic100/jQuery-QueryBuilder/blob/master/.github/CONTRIBUTING.md)
- [X] I created my branch from `dev` and I am issuing the PR to `dev`
- [X] I didn't pushed the `dist` directory
- [X] Unit tests are OK (no unit tests)
- [X] If it's a new feature, I added the necessary unit tests  (no unit tests)
- [X] If it's a new language, I filled the `__locale` and `__author` fields (not applicable)

Hello,
First of all thanks for this awesome project!
I'm currently using it with the ``chosen`` plugin but I had to modify a bit the source code of the ``plugins/chosen-selectpicker/plugin.js`` script so that it better fits my needs.

Please, if you have any comments or suggestions about the code, feel free to share them.

## Changes

- Avoid creating a useless glitchy _chosen_ widget when the operator only contains one option
- Add a _chosen_ widget on the value container (if eligible)

### CODE REVIEW

File: src/plugins/chosen-selectpicker/plugin.js [hunk 1]
```diff
@@ -23,7 +23,15 @@ QueryBuilder.define('chosen-selectpicker', function(options) {
     });
 
     this.on('afterCreateRuleOperators', function(e, rule) {
-        rule.$el.find(Selectors.rule_operator).removeClass('form-control').chosen(options);
+        var node = rule.$el.find(Selectors.rule_operator);
+        if (node.css('display') !== 'none') { // avoid creating useless chosen widget
```
Comment (mistic100) [inline src/plugins/chosen-selectpicker/plugin.js]: `self.getOperators(rule.filter).length > 1` would be a better test

File: src/plugins/chosen-selectpicker/plugin.js [hunk 2]
```diff
@@ -23,7 +23,15 @@ QueryBuilder.define('chosen-selectpicker', function(options) {
...
+            rule.$el.find(Selectors.rule_operator).removeClass('form-control').chosen(options);
+        }
+    });
+
+    this.on('afterCreateRuleInput', function(e, rule) {
```
Comment (mistic100) [inline src/plugins/chosen-selectpicker/plugin.js]: Builder plugins are not to be applied on value inputs, this is the purpose of the `filter.plugin` options, please remove this chunk

File: src/plugins/chosen-selectpicker/plugin.js (hunk 1 above)
Comment (mokaddem) [reply src/plugins/chosen-selectpicker/plugin.js]: I guess you mean ``e.builder.getOperators(rule.filter).length > 1`` ?

File: src/plugins/chosen-selectpicker/plugin.js (hunk 2 above)
Comment (mokaddem) [reply src/plugins/chosen-selectpicker/plugin.js]: Wow. Completely missed that. Awesome!

**Pull request merged**